server/protocols/protocols-imap4/src/main/java/org/apache/james/imapserver/netty/IMAPServerFactory.java

```diff
@@ -76,16 +78,19 @@ protected IMAPServer createServer(HierarchicalConfiguration<ImmutableNode> confi
     protected List<AbstractConfigurableAsyncServer> createServers(HierarchicalConfiguration<ImmutableNode> config) throws Exception {
         List<AbstractConfigurableAsyncServer> servers = new ArrayList<>();
         List<HierarchicalConfiguration<ImmutableNode>> configs = config.configurationsAt("imapserver");
-        
+
         for (HierarchicalConfiguration<ImmutableNode> serverConfig: configs) {
             IMAPServer server = createServer(serverConfig);
             server.setFileSystem(fileSystem);
             server.configure(serverConfig);
             servers.add(server);
         }
+        imapServers = servers.stream().map(IMAPServer.class::cast).toList();

```

Comment:
What is this uggly cast?


Can't we `List<IMAPServer> servers` in the first place?